src/tools/asm2wasm.cpp

```diff
@@ -132,8 +140,7 @@ int main(int argc, const char *argv[]) {
   }
 
   Asm2WasmPreProcessor pre;
-  // wasm binaries can contain a names section, but not full debug info
-  pre.debugInfo = passOptions.debugInfo && !emitBinary;
+  pre.debugInfo = passOptions.debugInfo;
```

Comment:
if we are emitting binary, and not emitting a binary map, we can drop the debug info, I think? I.e. replace `!emitBinary` with something like `(!emitBinary || !binaryMapFile)`